### DISCUSSION TITLE
workouts

### DISCUSSION THREAD

**Discussion by vonfoxie87** [dvmarinoff/Auuki#162] [General]
i am running your awesome application on my own machine now, i am very pleased with it.

Besides the fact i am very happy, i am wondering why its not possible to just drop .zwo files into the workouts directory. I wish to use many .zwo workouts and its not really usefull to edit the js file with al those workouts. 

Is there a workaround to make this happen?

Thanks.

**Comment by dvmarinoff** [comment]
Hi and welcome to the repo!

I've just deliberated on a few possible ways to implement a similar functionality and I think the best approach given all the possible use cases is to put the workouts storage in IndexedDB in the browser. Will sit down to implement it over the weekend and will take into account the bulk upload too.

**Comment by vonfoxie87** [comment]
Thank you in advance.

I was looking into it myself but I keep breaking it up. Js isn't my language unfortunately. I wish I new how to make this with python, I can't find any documentation about it. Well done to you for figuring this out.

**Comment by dvmarinoff** [comment]
Hey just to let you know I started work on that issue and it will be available for preview on the develop branch which is hosted here: https://flux-dev.vercel.app/

Currently I managed to only add the ability to store the User added workouts to IndexedDB. Those will be merged with the default once from the `.js` file. There is also a delete button, but I am going to rework that to be a swipe left to delete, later on. I also want to add a drag and drop input area, where whole folders could be added. This will take some more time to do, maybe 2 more weekends. I'll let you know when the work is stable and ready to be tested.

**Comment by vonfoxie87** [comment]
Thanks for all the hard work so far. Tomorrow I will try it out.

With using this style of dB, it is saved in the browsers history or does it work another way?
For example if I want sometime to use my phone or pc, I need to upload it at both devices?

**Comment by dvmarinoff** [comment]
Good news, finally had some time to sit down and finish this feature. It's on the develop branch and practically done.
Only a few little things wound remain to be polished with time. I am going to publish it on master later today.


https://github.com/dvmarinoff/Flux/assets/7523832/00f62f3a-d7dc-4b91-a6dc-0d1ded41b142

**Comment by dvmarinoff** [comment]
@vonfoxie87 the feature is now live on master and on the main site
